test/mocha/comments.js

```diff
@@ -260,6 +260,30 @@ describe("comments", function() {
         ].join("\n"));
     });
 
+    it("Should not duplicate a same comment", function() {
+        var result = UglifyJS.minify(
+`var foo = {};
+// this is a comment line
+(foo.bar = {}).test = 123;`, {
+            output: { comments: true }
+        });
+        if (result.error) throw result.error;
+        assert.strictEqual(result.code,
+`var foo={};
+// this is a comment line
+(foo.bar={}).test=123;`);
+        result = UglifyJS.minify(
+`var foo = {};
+/* this is a block line */
+(foo.bar = {}).test = 123;`, {
+            output: { comments: true }
+        });
+        if (result.error) throw result.error;
+        assert.strictEqual(result.code, 
+`var foo={};
+/* this is a block line */(foo.bar={}).test=123;`);
+    });
+
```

Comment:
I'm not happy to make this a mocha test. Terser output can change at will.

Can you make this a compress test? An example of a compress test ensuring that comments are as expected is this one: https://github.com/terser-js/terser/blob/master/test/compress/pure_funcs.js#L297-L313